File: python/door-problem/door_problem.py
import numpy as np
import copy
import matplotlib
import matplotlib.pyplot as plt


def toggleDoors(n):
    doors = -np.ones(n)
    allDoors = [copy.copy(doors)]
    for i in range(n):
        doors[range(i, n, i + 1)] *= -1
        allDoors.append(copy.copy(doors))
    return np.array(allDoors)


def main():
    doors = toggleDoors(100)
    plt.imshow(doors)
    plt.show()


if __name__ == "__main__":
    main()

File: julia/door-problem/door_problem.jl
vec2mat(v) = hcat(v...) |> transpose |> Matrix

function toggleDoors(n::Int)
	doors = -ones(Int,n)
	allDoors = [ copy(doors) ]

	for i in 1:n
		println(doors)
		doors[i:i:n] *= -1
		push!(allDoors, copy(doors))
	end

	return vec2mat(allDoors)
end

n = 15
doors = toggleDoors(n)

using Plots
heatmap(doors; legend=false)




